### PR TITLE
feat: support `Create Table ... Like`

### PR DESCRIPTION
**Pull request by KKould**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Support: `create table xxx like yyy`
ref: https://dev.mysql.com/doc/refman/8.0/en/create-table-like.html

```shell
public=> CREATE TABLE t1 ( a INT, b STRING, c INT, ts timestamp TIME INDEX);
OK 0
public=> CREATE TABLE t2 LIKE t1;
OK 0
public=> DESCRIBE t2;
 Column |         Type         | Key | Null | Default | Semantic Type 
--------+----------------------+-----+------+---------+---------------
 a      | Int32                |     | YES  |         | FIELD
 b      | String               |     | YES  |         | FIELD
 c      | Int32                |     | YES  |         | FIELD
 ts     | TimestampMillisecond | PRI | NO   |         | TIMESTAMP
(4 rows)
```

Tips: 
I was using this case: `CREATE TABLE t1 ( a INT, b STRING, c INT, ts timestamp TIME INDEX);` then run `CREATE TABLE t2 LIKE t1;`, the `create_partitions_stmt` method returned Some and caused the index to go out of bounds, so I did additional checks. Is this a bug? Because this table does not specify `Partitions`
bebd5cede7cdff3aef71baa6efbb7a2

ref: https://github.com/GreptimeTeam/greptimedb/pull/3372/files#diff-2563d4135f7e232aaec6a39204e6e9afc21801dff0c957772ce4dd7d3bd271d4R109

## Checklist

- [x]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.
- [x]  This PR does not require documentation updates.

## Refer to a related PR or issue link (optional)
https://github.com/GreptimeTeam/greptimedb/issues/2509

### CODE REVIEW

File: src/operator/src/expr_factory.rs [hunk 1]
```diff
@@ -181,7 +181,7 @@ pub(crate) async fn create_external_expr(
     Ok(expr)
 }
 
-/// Convert `CreateTable` statement to `CreateExpr` gRPC request.
+/// Convert `CreateTable` statement to `CreateTableExpr` gRPC request.
```
Comment (killme2008) [inline src/operator/src/expr_factory.rs:184]: Good catch! 👍

File: src/operator/src/statement/ddl.rs [hunk 2]
```diff
@@ -74,6 +78,47 @@ impl StatementExecutor {
...
+            .find_table_partitions(table_ref.table_info().table_id())
+            .await
+            .context(error::FindTablePartitionRuleSnafu { table_name: table })?;
+
+        let quote_style = ctx.quote_style();
+        let mut create_stmt =
+            show_create_table::create_table_stmt(&table_ref.table_info(), quote_style)
+                .context(error::ParseQuerySnafu)?;
+        create_stmt.name = stmt.name;
+        create_stmt.if_not_exists = false;
```
Comment (killme2008) [inline src/operator/src/statement/ddl.rs:107]: Looks like `create table like` can support `if_not_exists` too, just like

```sql
CREATE TABLE IF NOT EXISTS snapshot LIKE source_data;
```

File: src/sql/src/statements/create.rs [hunk 3]
```diff
@@ -218,6 +218,13 @@ pub struct CreateExternalTable {
     pub engine: String,
 }
 
+#[derive(Debug, PartialEq, Eq, Clone, Visit, VisitMut)]
+pub struct CreateTableLike {
+    /// Table name
+    pub name: ObjectName,
+    pub target: ObjectName,
```
Comment (killme2008) [inline src/sql/src/statements/create.rs:225]:
```suggestion
    /// The target table name
    pub target: ObjectName,
```

**Review by killme2008 — COMMENTED** [review]
Great work! Thank you. I have a few suggestions, please take a look.

File: src/operator/src/expr_factory.rs (hunk 1 above)
Comment (waynexia) [reply src/operator/src/expr_factory.rs:184]: Sugg: also add `[]` to those types so the language server can process them correctly

File: tests/cases/standalone/common/create/create.sql [hunk 4]
```diff
@@ -57,3 +57,17 @@ CREATE TABLE neg_default_value(i INT DEFAULT -1024, ts TIMESTAMP TIME INDEX);
 desc TABLE neg_default_value;
 
 DROP TABLE neg_default_value;
+
+CREATE TABLE test_like_1 (i INTEGER, j TIMESTAMP TIME INDEX);
```
Comment (waynexia) [inline tests/cases/standalone/common/create/create.sql:61]: Maybe make this case a bit more complicated, including default value, primary key, big letter identifier etc. The previous one could not recognize column constrain.

File: src/sql/src/statements/create.rs (hunk 3 above)
Comment (v0y4g3r) [reply src/sql/src/statements/create.rs:225]: The field name is a little bit confusing. `name` is actually the name of table to create, while `target` is where the schema comes from.

**Comment by KKould** [comment]
my grandmother passed away yesterday, and I will handle the review suggestions you provided in a few days

**Comment by killme2008** [comment]
> my grandmother passed away yesterday, and I will handle the review suggestions you provided in a few days

Sorry to hear that. May she rest in peace.

File: tests/cases/standalone/common/create/create.sql (hunk 4 above)
Comment (WenyXu) [reply tests/cases/standalone/common/create/create.sql:61]: +1

File: src/operator/src/statement/ddl.rs (hunk 2 above)
Comment (KKould) [reply src/operator/src/statement/ddl.rs:107]: In fact, I thought so at first, but I checked MySQL's `Create Table ... Like` and it seems that it is not supported, so I chose to keep it unified. Of course, this is up to you to decide.

ref: https://dev.mysql.com/doc/refman/8.0/en/create-table-like.html

**Comment by KKould** [comment]
It seems you didn’t notice the content in Tips. please take a look

File: tests/cases/standalone/common/create/create.sql (hunk 4 above)
Comment (KKould) [reply tests/cases/standalone/common/create/create.sql]: > Maybe make this case a bit more complicated, including default value, primary key, uppercase identifier etc. The previous one could not recognize column constraints.

whether secondary indexes are required, it seems that there is no support for the moment

File: src/operator/src/statement/ddl.rs (hunk 2 above)
Comment (killme2008) [reply src/operator/src/statement/ddl.rs:104]: Got it. Let's leave it currently.

**Review by killme2008 — APPROVED** [review]
LGTM

**Review by tisonkun — APPROVED** [review]
Thanks for your contribution @KKould! Excellent patch.

I suggest you try to check out a different branch (rather than `main`) the next time you submit a PR so that we can avoid some subtle issues like:

```
$ git push
fatal: The upstream branch of your current branch does not match
the name of your current branch.  To push to the upstream branch
on the remote, use

    git push https://github.com/KKould/greptimedb.git HEAD:main

To push to the branch of the same name on the remote, use

    git push https://github.com/KKould/greptimedb.git HEAD

To choose either option permanently, see push.default in 'git help config'.

To avoid automatically configuring an upstream branch when its name
won't match the local branch, see option 'simple' of branch.autoSetupMerge
in 'git help config'.
```

Although we can use `git push https://github.com/KKould/greptimedb.git HEAD:main` as suggested, it's error-prone.

**Pull request merged**